Makefile

```diff
@@ -120,6 +121,12 @@ e2e-test-native: ## Run end-to-end tests
 	@echo "Running e2e tests with CI=${CI}, APP_NAME=${APP_NAME}, BASE_URL=${BASE_URL}"
 	@cd e2e/$(APP_NAME) && APP_NAME=$(APP_NAME) BASE_URL=$(BASE_URL) npx playwright test $(E2E_ARGS)
 
+e2e-test-native-ui: ## Run Playwright UI tests natively
+	@:$(call check_defined, APP_NAME, You must pass in a specific APP_NAME)
+	@:$(call check_defined, BASE_URL, You must pass in a BASE_URL)
+	@echo "Running e2e UI tests natively with APP_NAME=$(APP_NAME), BASE_URL=$(BASE_URL)"
+	@cd e2e && APP_NAME=$(APP_NAME) BASE_URL=$(BASE_URL) npm run e2e-test:ui
```

Comment:
I feel like the e2e test running code needs to be polished. There are several issues in these files that should be polished up:

1. `e2e-test-native-ui` calls an npm script defined in package.json `npm run e2e-test:ui` while `e2e-test-native` calls playwright directly with `npx playwright`
2. the npm script `e2e-test` doesn't seem used anywhere
3. the number of layers of indirection from `make e2e-test-native-ui` to `npm run-e2e-test:ui` to `./run-e2e-test` and finally to `npx playwright` feels potentially excessive
4. the `UI=true` parameter to the `./run-e2e-test` script seems awkward as it's just a pass through to `--ui` option in npx playwright

can you suggest an approach that streamlines things and makes them more consistent

Reply:
personally, i can't think of a way to streamline, so maybe not possible. a couple of small improvements i can think of:

1. make things more consistent by calling `make e2e-test-native` call `npm run e2e-test`
2. make things more consistent by having `make e2e-setup-ci` call `npm run e2e-setup`
3. get rid of the Makefile from e2e/Dockerfile and just call `npm ci`, `npm run e2e-setup`, and `npm run e2e-test` directly

Reply:
Did the above updates

Reply:
I think providing the user flexibility of running from the root make (native and docker) and the e2e folder natively come with the tradeoff of it being overly complex. When you mentioned the levels of indirection above I wanted to see it visually - only things I can think of:
- Don't use npm commands at all and just use make to cut down on one level of indirection
- switch to a .env file for `APP_NAME` and `BASE_URL`
![image](https://github.com/user-attachments/assets/f385b2e2-495a-4d4b-8b49-0ec7e9bd993b)

Reply:
Thanks for making the changes, I think it's almost there, just a few more changes to keep things clean.

The Makefile shouldn't know about the core commands, we should abstract away all the core commands (e.g. playwright, etc) into npm scripts. This means:

- `make e2e-setup-native` should run `npm install` and `npm run e2e-setup` rather than `npx playwright install --with-deps` directly
- `make e2e-show-report` should run something like `npm run e2e-show-report`.

To summarize the abstractions at a high level:
- npm creates abstractions for running e2e commands and abstracts away the underlying library (playwright vs cypress vs selenium etc). we can change the underlying library and not change the npm scripts. everything is assumed to run natively at this level.
- Makefile creates abstractions for running e2e commands within Docker or natively. it has docker build steps, but under the hood it just calls npm scripts and doesn't know about the underlying e2e library.

Reply:
npm abstracting away the tool makes sense - I made what I think are the rest of the updates but lmk.

Even though the run-e2e-test script has playwright specific calls - technically that could use an npm command for `npx playwright test` . Cypress has a UI mode so all playwright references could be taken out of that file